<?xml version="1.0"?>
<launch>

	<!-- arg to config the launch file-->
        <!-- GRIPPER PARAM -->
        <arg name="robotiq_gripper"   default="false"/>

        <!-- COLLISION WALLS PARAM -->
        <arg name="collision_walls"   default="false"/>

	<arg name="launch_rviz_nav" default="true"/>
	<arg name="gazebo_world" default="$(find icclab_summit_xl)/worlds/summit_xl_office.world"/>
	<arg name="debug" default="false"/>
	<arg name="gui" default="true"/>
	<arg name="ros_planar_move_plugin" default="false"/>

	<arg name="id_robot_a" default="summit_xl"/>
	<arg name="x_init_pose_robot_a" default="0" />
	<arg name="y_init_pose_robot_a" default="0" />
	<arg name="z_init_pose_robot_a" default="0" />
	<arg name="localization_robot_a" default="true"/>
	<arg name="gmapping_robot_a" default="false"/>
	<arg name="move_base_robot_a" default="false"/>
	<arg name="amcl_and_mapserver_a" default="false"/>
	<arg name="map_file_a" default="willow_garage/willow_garage.yaml"/>
	<arg name="xacro_robot_a" default="irlab_summit_xls_std.urdf.xacro"/>
	<arg name="nav_test" default="false"/>
	<arg name="slam" default="false"/>
	<arg name="omni" default="true"/>


	<include file="$(find icclab_summit_xl)/launch/sim/irlab_summit_xl_one_robot.launch">
		<arg name="id_robot" value="$(arg id_robot_a)"/>
		<arg name="x_init_pose" value="$(arg x_init_pose_robot_a)"/>
		<arg name="y_init_pose" value="$(arg y_init_pose_robot_a)"/>
		<arg name="z_init_pose" value="$(arg z_init_pose_robot_a)"/>
		<arg name="xacro_robot" value="$(arg xacro_robot_a)"/>
		<arg name="launch_robot_localization" value="$(arg localization_robot_a)"/>
		<arg name="launch_amcl_and_mapserver" value="$(arg amcl_and_mapserver_a)"/>
		<arg name="launch_gmapping" value="$(arg gmapping_robot_a)"/>
		<arg name="map_file" value="$(arg map_file_a)"/>
		<arg name="launch_move_base" value="$(arg move_base_robot_a)"/>
		<arg name="ros_planar_move_plugin" value="$(arg ros_planar_move_plugin)"/>
		<arg name="nav_test" value="$(arg nav_test)"/>
		<arg name="slam" value="$(arg slam)"/>
		<arg name="omni" value="$(arg omni)"/>
		<arg name="robotiq_gripper" value="$(arg robotiq_gripper)"/>
		<arg name="collision_walls" value="$(arg collision_walls)"/>
	</include>

	<!-- gazebo simulation and rviz visualization-->
	<include file="$(find icclab_summit_xl)/launch/sim/irlab_gazebo_rviz.launch">
		<arg name="launch_rviz_nav" value="$(arg launch_rviz_nav)"/>
		<arg name="world" value="$(arg gazebo_world)"/>
		<arg name="debug" value="$(arg debug)"/>
		<arg name="gui" value="$(arg gui)"/>
	</include>

</launch>	
